### PR TITLE
Fix bdb loss log bug

### LINKED ISSUE #1587: Fe checkpoint failed: Failed to find key 25772638 in database 25770294
### Steps to reproduce the behavior (Required)
Possible in this scenario: restarting the master node in case of high metadata write pressure

### Real behavior (Required)
image

### StarRocks version (Required)
 - 1.18.2

### PR DESCRIPTION
**Pull request by gengjun-git**
Try to fix #1587 
For bdb je, Database.count() is not accurate. Get the last item in the database as the max journal id in the database.

### CODE REVIEW

**Comment by imay** [comment]
> For bdb je, Database.count() is not accurate. Get the last item in the database as the max journal id in the database.

Better to provide more information.
In what scenario, this will cause problem.

And explain more about your method.

**Comment by gengjun-git** [comment]
I will do some test before commit this PR.

File: fe/fe-core/src/main/java/com/starrocks/journal/bdbje/BDBJEJournal.java [hunk 1]
```diff
@@ -108,16 +108,14 @@ public synchronized void rollJournal() {
...
-        long newNameVerify = currentName + currentJournalDB.getDb().count();
+        long newNameVerify = currentJournalDB.getMaxJournalId() + 1;
         if (newName == newNameVerify) {
             LOG.info("roll edit log. new db name is {}", newName);
             currentJournalDB = bdbEnvironment.openDatabase(Long.toString(newName));
         } else {
-            String msg = String.format("roll journal error! journalId and db journal numbers is not match. "
-                            + "journal id: %d, current db: %s, expected db count: %d",
-                    newName, currentDbName, newNameVerify);
+            String msg = String.format("roll journal error! nestJournalId and db max journal id is not match. "
```
Comment (chaoyli) [inline fe/fe-core/src/main/java/com/starrocks/journal/bdbje/BDBJEJournal.java:116]: what does the nestJournalId meaning?

Comment (fireflyhoo) [reply fe/fe-core/src/main/java/com/starrocks/journal/bdbje/BDBJEJournal.java:116]: new journal id ?

Comment (gengjun-git) [reply fe/fe-core/src/main/java/com/starrocks/journal/bdbje/BDBJEJournal.java]: nextJournalId

**Pull request closed without merging**